### PR TITLE
Baseline Temporal Augmentation

### PR DESCRIPTION
**Pull request by joos2010kj**
Implementation of window slice

- works in progress: std/mean for Germany

### CODE REVIEW

**Review by ivanzvonkov — APPROVED** [review]

**Comment by ivanzvonkov** [comment]
Looks good, just might need some extra code to do window slice at test time

**Comment by joos2010kj** [comment]
Ran a test after merging with main and it's working fine with germany

**Comment by joos2010kj** [comment]
> Looks good, just might need some extra code to do window slice at test time

WIP

**Comment by ivanzvonkov** [comment]
Can this be merged?

File: src/utils/data_transform.py [hunk 1]
```diff
@@ -134,17 +144,25 @@ class PlanetTransform(EOTransformer):
...
     ):
         super().__init__(**kwargs)
+
+        if competition == "south_africa":
+            PlanetTransform.per_band_mean = np.array([580.4186, 852.98376, 1136.9423, 2761.0286])
+            PlanetTransform.per_band_std = np.array([179.95744, 209.66647, 384.34073, 476.9446])
+        elif competition == "germany":
+            PlanetTransform.per_band_mean = np.array([513.8759,  701.9154,  810.2146, 2778.312])
+            PlanetTransform.per_band_std = np.array([136.59517, 172.02454, 289.7672, 627.4169])
+
```
Comment (ivanzvonkov) [inline src/utils/data_transform.py]: Remove this in favor of simple scaling

File: submit.py [hunk 2]
```diff
@@ -143,11 +148,15 @@
 print(f"Storing the report at {name}.npz...")
 
 np.savez(
-    f"{name}.npz",
+    f"result/{name}.npz",
```
Comment (ivanzvonkov) [inline submit.py:151]: Saving as csv would make more sense with columns model_id, wandb_url, test accuracy, test loss

Comment (joos2010kj) [reply submit.py:151]: iirc, you already implemented changes to this in your commit, so I'll not update this part.

**Comment by joos2010kj** [comment]
To be merged after one last test after Slurm fix

File: src/utils/planet_reader.py [hunk 3]
```diff
@@ -48,31 +52,58 @@ def __init__(
 
         :return: None
         """
```
Comment (joos2010kj) [inline src/utils/planet_reader.py:54]: Please take a look closely at the changes for planet_reader.py, at this region, b/c I was not so familiar with your changes but there were some merge conflicts here, so I decided to just merge yours in (code is running fine w/ planetdaily).

File: src/utils/s1_s2_reader.py [hunk 4]
```diff
@@ -123,4 +126,23 @@ def __getitem__(self, idx):
...
+        Intify the timesteps (You need to add one more returned value to line 102: 
+        "x, y_true, mask, _ = batch" in train_valid_eval_utils.py)
+        return s1_s2_image_stack, s1_label, s1_s2_mask, s1_fid, np.array([int(f.strftime("%Y%m%d%H%M%S")) for f in timesteps])
+
+        Solution 2: < Selected
+        Not include timesteps since it does not show usefulness at the moment
+        return s1_s2_image_stack, s1_label, s1_s2_mask, s1_fid
+        """
+
+        return s1_s2_image_stack, s1_label, s1_s2_mask, s1_fid
```
Comment (joos2010kj) [inline src/utils/s1_s2_reader.py:148]: omitted timesteps for now

File: submit.py (hunk 2 above)
Comment (joos2010kj) [reply submit.py:151]: apparently it was a miscommunication, so I'll fix this

File: train.py [hunk 5]
```diff
@@ -24,102 +24,112 @@
...
 arg_parser = ArgumentParser()
-arg_parser.add_argument(
-    "--competition", type=str, default="south_africa", help="germany, south_africa"
-)
-arg_parser.add_argument("--model_type", type=str, default="spatiotemporal")
+
+# hyperparam
 arg_parser.add_argument("--batch_size", type=int, default=64)
 arg_parser.add_argument("--num_epochs", type=int, default=100)
+arg_parser.add_argument("--lr", type=float, default=0.001 if preference == "ivan" else 0.00001)
```
Comment (joos2010kj) [inline train.py:44]: Since you are using germany dataset often from now on, make sure to use 0.00001 for baseline, not 0.001 for LR

File: train.py [hunk 6]
```diff
@@ -24,102 +24,112 @@
...
+arg_parser.add_argument("--lr_scheduler", type=str, default="none")
+arg_parser.add_argument("--optimizer", type=str, default="Adam")
+arg_parser.add_argument("--loss", type=str, default="CrossEntropyLoss")
+
+# param
+arg_parser.add_argument(
+    "--competition", 
+    type=str, 
+    default="south_africa" if preference == "ivan" else "germany",
```
Comment (joos2010kj) [inline train.py:53]: By default, SA for you still.

File: train.py [hunk 7]
```diff
@@ -24,102 +24,112 @@
...
     type=str,
-    default="sentinel_2",
+    default="planet_daily",
     help="sentinel_1, sentinel_2, or planet_5day, s1_s2, planet_daily, s1_s2_planet_daily",
 )
 arg_parser.add_argument(
-    "--pos", type=str, default="both_34", help="both_34, 34S_19E_258N, 34S_19E_259N, 33N_18E_242N"
+    "--pos", 
+    type=str, 
+    default="both_34" if preference == "ivan" else "33N_18E_242N",
```
Comment (joos2010kj) [inline train.py:65]: Update the location if using Germany

**Review by joos2010kj — COMMENTED** [review]
@ivanzvonkov Please check this big PR!

File: create_csv.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
+import geopandas as gpd
```
Comment (ivanzvonkov) [inline create_csv.py:1]: Why is this an entirely new file? Isn't it already in main?

File: data_overview.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
+from pathlib import Path
+from tqdm import tqdm
```
Comment (ivanzvonkov) [inline data_overview.py:2]: ditto comment as above

File: data_overview.txt [hunk 10]
```diff
@@ -0,0 +1,27 @@
+/cmlscratch/izvonkov/tum-planet-radearth-ai4food-challenge/data
```
Comment (ivanzvonkov) [inline data_overview.txt:1]: Same comment as above

**Comment by ivanzvonkov** [comment]
I don't understand what happened here, it seems that a lot of the changes already on the main branch are being introduced again? Do you know what is going on?
